lightning/src/chain/keysinterface.rs

```diff
@@ -612,7 +612,8 @@ impl BaseSign for InMemorySigner {
 		for htlc in commitment_tx.htlcs() {
 			let htlc_tx = chan_utils::build_htlc_transaction(&commitment_txid, commitment_tx.feerate_per_kw(), self.holder_selected_contest_delay(), htlc, self.opt_anchors(), &keys.broadcaster_delayed_payment_key, &keys.revocation_key);
 			let htlc_redeemscript = chan_utils::get_htlc_redeemscript(&htlc, self.opt_anchors(), &keys);
-			let htlc_sighash = hash_to_message!(&bip143::SigHashCache::new(&htlc_tx).signature_hash(0, &htlc_redeemscript, htlc.amount_msat / 1000, SigHashType::All)[..]);
+			let htlc_sighashtype = if self.opt_anchors() { SigHashType::SinglePlusAnyoneCanPay } else { SigHashType::All };
+			let htlc_sighash = hash_to_message!(&bip143::SigHashCache::new(&htlc_tx).signature_hash(0, &htlc_redeemscript, htlc.amount_msat / 1000, htlc_sighashtype)[..]);
```

Comment:
Uhm, this is insecure, right? SINGLE|ANYONECANPAY on the counterparty commitment tx means our counterparty can remove all but the first output and replace them at will.

Reply:
Yea, BOLT 3 seems to say only use SINGLE|ANYONECANPAY on the HTLC transaction(s), which makes sense, as long as the outputs are sorted correctly.

Reply:
This spot in the code only signs the HTLC Success/Timeout tx, the commitment signature is generated above and uses `get_sighash_all`

Reply:
Oh, sorry, I'd forgotten we merged the HTLC and non-HTLC transaction signing into one function.